process/sync/baseSync.go

```diff
@@ -562,3 +565,41 @@ func isRandomSeedValid(header data.HeaderHandler) bool {
 
 	return !isPrevRandSeedNilOrEmpty && !isRandSeedNilOrEmpty
 }
+
+func (boot *baseBootstrap) requestHeadersFromNonceIfMissing(nonce uint64, hdrRes dataRetriever.HeaderResolver) {
+	var err error
+	nbRequestedHdrs := 0
+	maxNonce := core.MinUint64(nonce+maxHeadersToRequestInAdvance-1, boot.forkDetector.ProbableHighestNonce())
+	for currentNonce := nonce; currentNonce <= maxNonce; currentNonce++ {
+		if boot.shardCoordinator.SelfId() == sharding.MetachainShardId {
```

Comment:
I think it would be easier to understand it this verification is base on header type. ShardHeader / Metaheader.

Reply:
This func should be moved in the implementations (shard/meta) as to avoid this messy if.

Reply:
Refactored